eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -81,8 +81,9 @@ jobs:
 - ${{ if ne(parameters.ServiceDirectory, 'not-specified' )}}:
   - job: GenerateReleaseArtifacts
     pool:
-      vmImage: windows-2019
+      vmImage: $(OSVmImage)
     variables:
+      OSVmImage: windows-2019
```

Comment:
Does this get set in time? i.e. before the vmImage in the pool is evaulated?

Reply:
DevOps behaves as if this is the case but I don't see documentation describing that fact. I'm guessing that the `vmImage` is evaluating as expected here because it properly ingests values set in a matrix (https://github.com/Azure/azure-sdk-for-cpp/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L59) and matrix expansion tends to happen pretty late in the process